### PR TITLE
json rpc: de-dup intermediate digests in query_transaction_blocks

### PR DESCRIPTION
**Pull request by gegaowp**
## Description 

title

## Test plan 

CI with added sim test

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: crates/sui-json-rpc/src/indexer_api.rs [hunk 1]
```diff
@@ -217,6 +217,10 @@ impl<R: ReadApiServer> IndexerApiServer for IndexerApi<R> {
                 )
                 .await
                 .map_err(Error::from)?;
+            // De-dup digests, duplicate digests are possible, for example,
+            // when get_transactions_by_move_function with module or function being None.
+            let mut seen = std::collections::HashSet::new();
```
Comment (amnn) [inline crates/sui-json-rpc/src/indexer_api.rs:222]: nit: import `HashSet`?

**Review by amnn — APPROVED** [review]
I'm a little bit dubious that CI would tell you whether the fix was correct -- so it's worth double checking with a manual test, but the fix seems good to me, thanks for the quick turnaround @gegaowp !

**Comment by gegaowp** [comment]
@amnn I meant CI would protect other un-expected breaking, I added a sim test for this fix :)

File: crates/sui-json-rpc-tests/tests/transaction_tests.rs [hunk 2]
```diff
@@ -1,15 +1,25 @@
...
+use move_core_types::identifier::Identifier;
+use sui_json::{call_args, type_args};
 use sui_json_rpc_types::SuiTransactionBlockResponseQuery;
 use sui_json_rpc_types::TransactionFilter;
 use sui_json_rpc_types::{
     SuiObjectDataOptions, SuiObjectResponseQuery, SuiTransactionBlockResponse,
     SuiTransactionBlockResponseOptions, TransactionBlockBytes,
 };
 use sui_macros::sim_test;
+use sui_types::base_types::ObjectID;
```
Comment (gegaowp) [inline crates/sui-json-rpc-tests/tests/transaction_tests.rs:15]: these are just cargo fmt changes.

**Pull request merged**